### PR TITLE
Source Bing Ads: add integration tests 

### PR DESCRIPTION
**Pull request by darynaishchenko**
resolved: https://github.com/airbytehq/airbyte-internal-issues/issues/6654

Added integration tests for streams:
    1. app_install_ads
    2. app_install_ad_labels
    3. age_gender_audience_report_hourly
    4. account_impression_performance_report_hourly
    5. keyword_performance_report_hourly
    6. ad_performance_report_hourly
    7. ad_group_impression_performance_report_hourly
    8. campaign_performance_report_hourly
    9. campaign_impression_performance_report_hourly
    10. geographic_performance_report_hourly
    11. search_query_performance_report_hourly
    12. user_location_performance_report_hourly
    13. ad_group_performance_report_hourly
    14. account_performance_report_hourly
    
*_hourly_reports cannot be tested with cat as sync is too long.   app_install_ads and app_install_ad_labels streams are empty.
Also updated expected records with *monthly reports and removed them from cat cinfig.

### CODE REVIEW

**Review by maxi297 — APPROVED** [review]
Conditional approval on improving the maintainability of 
* airbyte-integrations/connectors/source-bing-ads/unit_tests/integrations/test_app_install_ad_labels_stream.py
* airbyte-integrations/connectors/source-bing-ads/unit_tests/integrations/test_budget_stream.py

File: airbyte-integrations/connectors/source-bing-ads/unit_tests/integrations/test_app_install_ad_labels_stream.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        self.auth_client(http_mocker)
+        output, _ = self.read_stream(self.stream_name, SyncMode.full_refresh, self._config, "app_install_ad_labels")
+        assert len(output.records) == 1
+
+    @HttpMocker()
+    def test_return_logged_info_for_empty_csv_file(self, http_mocker: HttpMocker):
+        self.auth_client(http_mocker)
+        output, _ = self.read_stream(self.stream_name, SyncMode.full_refresh, self._config, "app_install_ad_labels_empty")
+        assert len(output.records) == 0
+        assert len(output.logs) == 10
```
[the remark this answers is not in the document]
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-bing-ads/unit_tests/integrations/test_app_install_ad_labels_stream.py:25]: we discussed this in previous pr https://github.com/airbytehq/airbyte/pull/35201#discussion_r1496401275

File: airbyte-integrations/connectors/source-bing-ads/unit_tests/integrations/test_app_install_ad_labels_stream.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        state = self._state("app_install_ad_labels_state", self.stream_name)
+        self.auth_client(http_mocker)
+        output, service_call_mock = self.read_stream(
+            self.stream_name,
+            SyncMode.incremental,
+            self._config,
+            "app_install_ad_labels_with_state",
+            state
+        )
+        assert len(output.records) == 4
```
[the remark this answers is not in the document]
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-bing-ads/unit_tests/integrations/test_app_install_ad_labels_stream.py:55]: we don't filter records in this streams, so number of records depends of response, I will remove this assert

File: airbyte-integrations/connectors/source-bing-ads/unit_tests/integrations/test_budget_stream.py [hunk 3]
```diff
@@ -38,7 +38,7 @@ def test_incremental_read_cursor_value_matches_value_from_most_recent_record(sel
         self.auth_client(http_mocker)
         output, _ = self.read_stream(self.stream_name, SyncMode.incremental, self._config, "budget_with_cursor_value")
         assert len(output.records) == 8
-        assert output.most_recent_state.get(self.stream_name, {}).get(self.account_id, {}) == {self.cursor_field: "2024-01-01T12:54:12.028+00:00"}
+        assert output.most_recent_state.get(self.stream_name, {}).get(self.account_id, {}) == {self.cursor_field: "2024-03-01T12:54:12.028+00:00"}
```
[the remark this answers is not in the document]
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-bing-ads/unit_tests/integrations/test_budget_stream.py]: I reverted this changes, thanks

**Review by lazebnyi — APPROVED** [review]

**Pull request merged**